modules/fleet-membership/main.tf

```diff
@@ -15,8 +15,9 @@
  */
 
 locals {
-  hub_project_id          = var.hub_project_id == "" ? var.project_id : var.hub_project_id
-  gke_hub_membership_name = var.membership_name != "" ? var.membership_name : "${var.project_id}-${var.location}-${var.cluster_name}"
+  hub_project_id                   = var.hub_project_id == "" ? var.project_id : var.hub_project_id
+  gke_hub_membership_name_complete = var.membership_name != "" ? var.membership_name : "${var.project_id}-${var.location}-${var.cluster_name}"
+  gke_hub_membership_name          = length(local.gke_hub_membership_name_complete) > 63 ? substr(local.gke_hub_membership_name_complete, 0, 63) : local.gke_hub_membership_name_complete
```

Comment:
nit: we may not need the len check since  substr(string,0,n>len(string) == string anyway.

Reply:
Also looks like it per API docs it must end with an alphanumeric character. Let's proactively wrap a trimsuffix(substr..,"-") if for some reason the result of substr turns out to end in "-".

Reply:
fixed